### PR TITLE
Feature/1 move identifier from rls

### LINKED ISSUE #1: Move Identifier from RLS
## Situation

During the course of writing exasol/row-level-security, we created an object and an interface to represent database identifiers. This object contains validity checks applied before creating an identifier, so that identifiers are valid according to the Exasol SQL rules and can't be used for SQL injection.

## Acceptance Criteria

1. Identifier code extracted from RLS

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #1.

### CODE REVIEW

File: .classpath [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AnastasiiaSergienko) [inline .classpath:1]: Do you need this file?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>integration-test</goal>
+                            <goal>verify</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
```
Comment (AnastasiiaSergienko) [inline pom.xml:148]: gpg plugin is missing

File: doc/user_guide/user_guide.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Exasol Database Fundamentals
+
+This project contains a library that offers support for the most basic building blocks of the Exasol database, like [identifiers](https://intranet.srv.exasol.com/display/TOOLTIPS/Travis+CI+Invoices) and [keywords](). It also implements rules that the Exasol database follows &mdash; for example quoting. 
+
+## Introduction
+
+Each database product has a set of defining traits, some of which stem from the SQL standard, whereas others are database-specific. 
+
+Identifiers for example are a case where sources apply. The Standard defines general limitations on identifiers, e.g. that an unquoted identifier cannot contain spaces. Additionally, Exasol adds rules what characters can an cannot be used in identifiers.
```
Comment (AnastasiiaSergienko) [inline doc/user_guide/user_guide.md]:
```suggestion
Identifiers for example are a case where sources apply. The Standard defines general limitations on identifiers, e.g. that an unquoted identifier cannot contain spaces. Additionally, Exasol adds rules what characters can and cannot be used in identifiers.
```

File: doc/user_guide/user_guide.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+The main purpose of this library is to make sure that Java clients accessing Exasol don't need to individually implement the base objects and ground rules, but instead can rely on a ready-made API that makes sure, those things are handled correctly.
+
+## Features
+
+### Ensuring Identifier Validity
+
+One of the most often used base object in any database is the identifier. Identifiers are names of database objects like schemas, tables, users and so on.
```
Comment (AnastasiiaSergienko) [inline doc/user_guide/user_guide.md]:
```suggestion
One of the most often used base objects in any database is the identifier. Identifiers are names of database objects like schemas, tables, users and so on.
```

File: src/main/java/com/exasol/db/Identifier.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.exasol.db;
+
+public interface Identifier {
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/db/Identifier.java:6]: JavaDoc?

**Review by AnastasiiaSergienko — CHANGES_REQUESTED** [review]

File: .classpath (hunk 1 above)
Comment (redcatbear) [reply .classpath:1]: Makes your life with Eclipse easier.

File: pom.xml (hunk 2 above)
Comment (redcatbear) [reply pom.xml:148]: Added. Also to README.

File: src/main/java/com/exasol/db/Identifier.java (hunk 5 above)
Comment (redcatbear) [reply src/main/java/com/exasol/db/Identifier.java:6]: Added.

**Review by AnastasiiaSergienko — APPROVED** [review]

**Pull request merged**